pkg/network/multus/multus.go

```diff
@@ -35,3 +37,12 @@ func FilterNonDefaultNetworks(networks []v1.Network) []v1.Network {
 	}
 	return multusNetworks
 }
+
+func LookUpDefaultNetwork(networks []v1.Network) (v1.Network, error) {
+	for _, network := range networks {
+		if network.Multus == nil || network.Multus.Default {
+			return network, nil
+		}
+	}
+	return v1.Network{}, fmt.Errorf("no multus default network found")
```

Comment:
Existing patterns in `vmispec` are returning a `nil` if nothing was found, which seems a better fit IMO.
It would be even better if the lookup will return the actual network (original one) and not a copy.

If someone needs to expose the failure to find a network, then that caller should raise an error.

Reply:
I was struggling with myself on that, and finally decided to have it return a non pointer for unit-test reasons.. I have no problem to change it back..

Reply:
If it returns the original network, then the caller can mutate it in-place if it needs to. If it returns a copy, it needs to override the original one which I find less attractive.

In tests you may even check identity instead of equality.

Reply:
yep I agree.